web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/index.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { useWalletClient, useAccount } from "wagmi";
+import { Button } from "@kleros/ui-components-library";
+import { uploadSettingsToSupabase } from "utils/uploadSettingsToSupabase";
+import FormContact from "./FormContact";
+import messages from "../../../../../../../consts/eip712-messages";
+import { ISettings } from "../../types";
+
+const FormContainer = styled.form`
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  padding: 0 calc(12px + (32 - 12) * ((100vw - 300px) / (1250 - 300)));
+  padding-bottom: 16px;
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: end;
+`;
+
+const FormContactContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin-bottom: 24px;
+`;
+
+const FormContactDetails: React.FC<ISettings> = ({ setIsSettingsOpen }) => {
+  const [telegramInput, setTelegramInput] = useState<string>("");
+  const [emailInput, setEmailInput] = useState<string>("");
+  const [telegramIsValid, setTelegramIsValid] = useState<boolean>(false);
+  const [emailIsValid, setEmailIsValid] = useState<boolean>(false);
+  const { data: walletClient } = useWalletClient();
+  const { address } = useAccount();
+
+  // TODO: after the user is authenticated, retrieve the current email/telegram from the database and populate the form
```

Comment:
I was curious about how are we going to retrieve the currently saved email/telegram from the database and populate the form without leaking information. For example you can impersonate being connected on any web3 wallet (e.g. Vitalik's), which would leak their telegram handle & email address

Reply:
"after the user is authenticated", which could be a signature, only then we would retrieve the contact details from DB. But asking for too many signatures is bad, we should do it once and store. At this point we'd be better off using [SSX](https://www.sprucekit.dev/ssx/ssx-overview) which implements a standard SIWE

Reply:
Follow-up -> https://github.com/kleros/kleros-v2/issues/1264